### PR TITLE
Support ElasticSearch 8 and add it into E2E tests

### PR DESCRIPTION
**Pull request by kezhenxu94**
- [x] If this pull request closes/resolves/fixes an existing issue, replace the issue number. Closes #<issue number>.
- [x] Update the [`CHANGES` log](https://github.com/apache/skywalking/blob/master/CHANGES.md).

### CODE REVIEW

**Comment by wu-sheng** [comment]
`backend-storage.md` should be updated if tests passed.

File: oap-server/server-storage-plugin/storage-elasticsearch-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/elasticsearch/base/StorageEsInstaller.java [hunk 1]
```diff
@@ -215,9 +215,9 @@ protected Mappings createMapping(Model model) {
                 properties.put(matchCName, matchColumn);
             } else {
                 Map<String, Object> column = new HashMap<>();
-                column.put(
-                    "type", columnTypeEsMapping.transform(columnDefine.getType(), columnDefine.getGenericType()));
-                if (columnDefine.isStorageOnly()) {
+                column.put("type", type);
+                // no index parameter is allowed for binary type, since ES 8.0
```
Comment (wu-sheng) [inline oap-server/server-storage-plugin/storage-elasticsearch-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/elasticsearch/base/StorageEsInstaller.java:219]: Curiously, What binary fields are we setting `parameter`? And which parameters did we set?

Comment (kezhenxu94) [reply oap-server/server-storage-plugin/storage-elasticsearch-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/elasticsearch/base/StorageEsInstaller.java:219]: Parameter name is "index", examples are "tags"

Comment (kezhenxu94) [reply oap-server/server-storage-plugin/storage-elasticsearch-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/elasticsearch/base/StorageEsInstaller.java:219]: https://github.com/apache/skywalking/blob/f7bcb76784f968b865f31c7abfbd7c3a33945986/oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/alarm/AlarmRecord.java#L80

Comment (wu-sheng) [reply oap-server/server-storage-plugin/storage-elasticsearch-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/elasticsearch/base/StorageEsInstaller.java:219]: Oh, I see. It actually should add `storageOnly = true`, this is also a code bug.

Comment (kezhenxu94) [reply oap-server/server-storage-plugin/storage-elasticsearch-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/elasticsearch/base/StorageEsInstaller.java:219]: > Oh, I see. It actually should add `storageOnly = true`, this is also a code bug.

I can fix it, but FYI, regardless of that even we set `index=false`, it's still not allowed in ES8 to have parameter `index` in binary type.

Comment (kezhenxu94) [reply oap-server/server-storage-plugin/storage-elasticsearch-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/elasticsearch/base/StorageEsInstaller.java:219]: > Oh, I see. It actually should add `storageOnly = true`, this is also a code bug.

Fixed in this patch.

**Review by wu-sheng — APPROVED** [review]

**Pull request merged**